src/views/templates/details/tabs/parameters/TemplateParametersPage.tsx

```diff
@@ -21,41 +31,39 @@ type TemplateParametersPageProps = RouteComponentProps<{
 };
 
 const TemplateParametersPage: React.FC<TemplateParametersPageProps> = ({ obj: template }) => {
+  const [editableTemplate, setEditableTemplate] = useImmer(template);
+
+  useEffect(() => setEditableTemplate(template), [setEditableTemplate, template]);
+
   const { t } = useKubevirtTranslation();
   const isEditDisabled = isCommonVMTemplate(template);
   const history = useHistory();
-  const [parameters, updateParameters] = useImmer<TemplateParameter[]>(template.parameters);
   const [error, setError] = React.useState();
   const [success, setSuccess] = React.useState(false);
   const [loading, setLoading] = React.useState(false);
 
-  const isSaveDisabled =
-    JSON.stringify(template.parameters) === JSON.stringify(parameters) || isEditDisabled;
-
   const onParameterChange = (parameter: TemplateParameter) => {
-    updateParameters((draftParameters) => {
+    setEditableTemplate(({ parameters: draftParameters }) => {
       const parameterIndex = draftParameters.findIndex((p) => p.name === parameter.name);
       draftParameters[parameterIndex] = parameter;
     });
   };
 
+  const parameters = editableTemplate.parameters;
+
+  const isSaveDisabled =
+    JSON.stringify(template.parameters) === JSON.stringify(parameters) || isEditDisabled;
```

Comment:
stringify can be sometimes a bit heavy, consider putting those lines of params and stringify in a usememo, as this component has a lot of states which will cause a lot or recalculate that arent needed if parmas isn't updated